### PR TITLE
Add const T to docstring generation.

### PR DESCRIPTION
**Pull request by jhale**
## Description

When using const T in pyarray auto-generated documentation shows long double type.

## Suggested changelog entry:

Fix auto-generated documentation string when using const T in pyarray_t.

### CODE REVIEW

**Comment by rwgk** [comment]
Hi @jhale, I think the valgrind error is unrelated (looks like pocketfft suppression needs to be updated again), but could you please add a test that fails without your change, and passes with your change?

**Comment by rwgk** [comment]
I fixed the valgrind error. If you rebase this PR on master the CI should come back clean.

**Comment by jhale** [comment]
I've added a test that fails on `master` and passes with my branch.

File: tests/test_numpy_array.py [hunk 1]
```diff
@@ -481,6 +481,9 @@ def test_index_using_ellipsis():
     a = m.index_using_ellipsis(np.zeros((5, 6, 7)))
     assert a.shape == (6,)
 
+def test_format_descriptors_for_const_types():
```
Comment (rwgk) [inline tests/test_numpy_array.py:484]: For completeness, could you please add corresponding non-const tests, even if redundant (I didn't look if that's covered elsewhere already). It's usually good to exercise closely related contrasting functionality together, especially if it's super easy, as in this case.

**Review by rwgk — COMMENTED** [review]
Could you please fix the Format issue? — I usually blindly run `pre-commit run --all-files`. See .github/CONTRIBUTING.md.

Please ignore the Centos 8 failure, that's a general breakage as of ~4 days ago.

File: tests/test_numpy_array.cpp [hunk 2]
```diff
@@ -437,4 +437,10 @@ TEST_SUBMODULE(numpy_array, sm) {
     sm.def("accept_double_f_style_forcecast_noconvert",
            [](py::array_t<double, py::array::forcecast | py::array::f_style>) {},
            "a"_a.noconvert());
+
+    // Check that types returns correct npy format descriptor
+    sm.def("float", [](py::array_t<float>) {});
```
[the remark this answers is not in the document]
Comment (rwgk) [reply tests/test_numpy_array.cpp]: I figured you're in a European time zone, so I went ahead and started working on the prefix. When I looked more closely I decided `test_fmt_desc_` is better as a prefix, and while I was at it I also condensed the test and made it less platform-specific.

**Comment by rwgk** [comment]
CI looks good, except for the Centos 8 build with is known to be broken for unrelated reasons. I'll merge this PR now. Thanks @jhale for the fix!

**Pull request merged**